usr/local/www/system_firmware_settings.php

```diff
@@ -86,196 +86,170 @@
-				<br />
-				<b><?=gettext("NOTE:"); ?></b> <?php printf(gettext("When a custom URL is configured, the system will not verify the image has an official digital signature")); ?>
-				</span>
-				</td>
-	</tr>
-	<tr>
-		<td colspan="2" class="list" height="12">&nbsp;</td>
-	</tr>
-	<tr>
-		<td colspan="2" valign="top" class="listtopic"><?=gettext("Updates"); ?></td>
-	</tr>
-	<tr>
-		<td width="22%" valign="top" class="vncell"><?=gettext("Unsigned images"); ?></td>
-		<td width="78%" class="vtable">
-			<input name="allowinvalidsig" type="checkbox" id="allowinvalidsig" value="yes" <?php if (isset($curcfg['allowinvalidsig'])) echo "checked=\"checked\""; ?> />
-			<br />
-			<?=gettext("Allow auto-update firmware images with a missing or invalid digital signature to be used."); ?>
-		</td>
-	</tr>
-	<tr>
-		<td width="22%" valign="top" class="vncell"><?=gettext("Dashboard check"); ?></td>
-		<td width="78%" class="vtable">
-			<input name="disablecheck" type="checkbox" id="disablecheck" value="yes" <?php if (isset($curcfg['disablecheck'])) echo "checked=\"checked\""; ?> />
-			<br />
-			<?=gettext("Disable the automatic dashboard auto-update check."); ?>
-		</td>
-	</tr>
-<?php if(file_exists("/usr/local/bin/git") && $g['platform'] == "pfSense"): ?>
-	<tr>
-		<td colspan="2" class="list" height="12">&nbsp;</td>
-	</tr>
-	<tr>
-		<td colspan="2" valign="top" class="listtopic"><?=gettext("Gitsync"); ?></td>
-	</tr>
-	<tr>
-		<td width="22%" valign="top" class="vncell"><?=gettext("Auto sync on update"); ?></td>
-		<td width="78%" class="vtable">
-			<input name="synconupgrade" type="checkbox" id="synconupgrade" value="yes" <?php if (isset($gitcfg['synconupgrade'])) echo "checked=\"checked\""; ?> />
-			<br />
-			<?=gettext("After updating, sync with the following repository/branch before reboot."); ?>
-		</td>
-	</tr>
-<?php
-	if(is_dir("/root/pfsense/pfSenseGITREPO/pfSenseGITREPO")) {
-		exec("cd /root/pfsense/pfSenseGITREPO/pfSenseGITREPO && git config remote.origin.url", $output_str);
-		if(is_array($output_str) && !empty($output_str[0]))
-			$lastrepositoryurl = $output_str[0];
-		unset($output_str);
+
+	require('classes/Form.class.php');
+
+	$form = new Form();
+
+	$section = new Form_Section('Firmware Branch');
+
+	if(is_array($preset_urls_split)) {
+		$urllist = array();
+
+		foreach($preset_urls_split as $pus) {
+			$pus_text = explode("\t", $pus);
+			if (empty($pus_text[0]))
+				continue;
+			if (stristr($pus_text[0], php_uname("m")) !== false) {
+				$yourarch = " (Current architecture)";
+				$choice = $pus_text[1];
+			} else {
+				$yourarch = "";
+			}
+
+			$urllist[$pus_text[1]] = $pus_text[0] . $yourarch;
+		}
+
+		$section->addInput(new Form_Select(
+		   'preseturls',
+		   'Default Auto Update URLs',
+		   $choice,
+		   $urllist
+		   ))->setHelp('Entries denoted by "Current architecture" match the architecture of your current installation, ' .
```

Comment:
For parameters in help-text; please use earlier described syntax:
```
setHelp('such as %s. Changing architectures...', [php_uname("m")]);
```
otherwise the translators would have to translate all different architectures ;) This is also needed for eg. *firmwareurl* and *branch*